src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/watasks/controllers/PostTaskSearchControllerTest.java

```diff
@@ -193,4 +196,103 @@ public void should_return_200_with_task_with_additional_properties_which_include
             .forEach(task -> common.cleanUpTask(task.getTaskId()));
     }
 
+    @Test
+    public void should_return_200_with_tasks_sorted_on_next_hearing_date_desc() {
+        common.setupCFTOrganisationalRoleAssignmentForWA(caseworkerCredentials.getHeaders());
+        List<TestVariables> tasksCreated = new ArrayList<>();
+
+        TestVariables taskVariables = common.setupWATaskAndRetrieveIds("requests/ccd/wa_case_data.json");
+        tasksCreated.add(taskVariables);
+        initiateTask(caseworkerCredentials.getHeaders(), taskVariables,
+                     "reviewSpecificAccessRequestLegalOps", "task name", "task title");
+
+        taskVariables = common.setupWATaskAndRetrieveIds("requests/ccd/wa_case_data.json");
+        tasksCreated.add(taskVariables);
+        initiateTask(caseworkerCredentials.getHeaders(), taskVariables,
+                     "reviewSpecificAccessRequestLegalOps", "task name", "task title");
+
+        List<String> taskIds = tasksCreated.stream().map(TestVariables::getTaskId).collect(Collectors.toList());
+        List<String> caseIds = tasksCreated.stream().map(TestVariables::getCaseId).collect(Collectors.toList());
+
+        SearchTaskRequest searchTaskRequest = new SearchTaskRequest(
+            asList(
+                new SearchParameterList(JURISDICTION, SearchOperator.IN, singletonList("WA")),
+                new SearchParameterList(LOCATION, SearchOperator.IN, singletonList("765324")),
+                new SearchParameterBoolean(AVAILABLE_TASKS_ONLY, SearchOperator.BOOLEAN, false),
+                new SearchParameterList(CASE_ID, SearchOperator.IN, caseIds)
+            ),
+            List.of(new SortingParameter(SortField.NEXT_HEARING_DATE_CAMEL_CASE, SortOrder.DESCENDANT))
+        );
+
+        tasksCreated.forEach(testVariable ->
+                                 common.insertTaskInCftTaskDb(testVariable, "processApplication", caseworkerCredentials.getHeaders()));
+
+        Response result = restApiActions.post(
+            ENDPOINT_BEING_TESTED + "?first_result=0&max_results=10",
+            searchTaskRequest,
+            caseworkerCredentials.getHeaders()
+        );
+
+        result.prettyPrint();
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.OK.value())
+            .body("tasks.size()", lessThanOrEqualTo(10)) //Default max results
```

Comment:
could we include the hearindDate on the task in the assertion? That way we firm up the attribute we have just sorted on.